packages/edit-site/src/components/layout/router.js

```diff
@@ -108,9 +108,7 @@ export default function useLayoutAreas() {
 		return {
 			key: 'templates-list',
 			areas: {
-				sidebar: postId ? (
```

Comment:
There's only one screen that uses the `/wp_template` path, as far as I'm aware: in that screen, whether we have a postId or not, we want to show the `SidebarNavigationScreenTemplatesBrowse`. Unless I'm missing something else?

Reply:
I've looked at the Sidebar before the router refactoring: https://github.com/WordPress/gutenberg/pull/60466/files#diff-0cf5246119f61f0f26cb94918697d2cb3a0f40051045d24dd01c98241e79bde7L79 and I see how that logic could have been confusing when we ported the logic to the router.

There was two places that used the `/wp_template`:

```jsx
<SidebarScreenWrapper path="/:postType(wp_template)">
    <SidebarNavigationScreenTemplatesBrowse />
</SidebarScreenWrapper>
// ...
<SidebarScreenWrapper path="/:postType(wp_template)/:postId">
    <SidebarNavigationScreenTemplate />
</SidebarScreenWrapper>
```

I haven't checked how the navigator worked, but it sounds like the second path would be never reached? I don't think we had a `/wp_template/postId` path.

Reply:
The second path would be reached when you were on a `site-editor.php?postType=wp_template&postId=x` page. Then, in the `useSyncPathWithURL` hook, there was a `getPathFromURL` that transformed these URL params into a Navigator path `/wp_template/x`. And that matched the `SidebarScreenWrapper` that shows `SidebarNavigationScreenTemplate`.

To get the `/:postType(wp_template)` path, you URL had to be `site-editor.php?path=/wp_template`. Then the Navigator displays `SidebarNavigationScreenTemplatesBrowse`, which is the "index".

So, yes, your fix is correct. When the URL is `site-editor.php?path=/wp_template&postId=x`, we're not on the editor page for template `x`, but we're on the template index page with `x` selected/highlighted.

Reply:
> The second path would be reached when you were on a site-editor.php?postType=wp_template&postId=x page

Nice, thanks for the confirmation. This is the "details page" and we have that already [covered in the router](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/layout/router.js#L87), so we're good. 👍